### ISSUE TITLE
xarray fields in scipy mode

### ISSUE THREAD

**Issue by delandmeterp** [OceanParcels/parcels#601]
Hi @willirath,

I’ve just started to remove explicit conversion from xr.array to np.array in Parcels. I want this stage only appearing in the c_library (then, in a second step, I’ll do this chunk by chunk).

So my idea is the following: I simply remove the numpy conversion in the NetcdfFileBuffer.

But then we do lots of operations on this array, such as 
```
            self.data[np.isnan(self.data)] = 0.
            if self.vmin is not None:
                self.data[self.data < self.vmin] = 0.
            if self.vmax is not None:
                self.data[self.data > self.vmax] = 0.
```
I remember that once you said we should not explicitly integrate dask in Parcels, but only xarray. So how can you do those operations with only xarray?

If it has to be done with dask, my idea was to simply do for u being a xarray:
```
u_dask = da.from_array(u, chunks=(1,2,100,100))  # the data we'll be using
u_dask[np.isnan(u_dask)] = 0.
u_dask[u_dask >  vmax] = 0.
u_dask[u_dask <  vmin] = 0.
```

And then, later on, when we know the the cell of the particle (xi, yi, zi, ti)
```
u_local = u_dask[ti:ti+2, zi:zi+2, yi:yi+2, xi:xi+2].compute()
```

Is there something wrong in this approach? So we need explicit usage of dask to do the lazy operations on the array, no?

**Comment by delandmeterp** [comment]
This is also necessary for periodic_halos:
```
ua = da.concatenate((ua, ua[:,:,:3,:]), axis=2)
```

**Comment by erikvansebille** [comment]
Closing, since this is now implemented in #605
